lib/widgets/home.dart

```diff
@@ -1,97 +1,539 @@
+  @override
+  void initState() {
+    super.initState();
+    _tab.addListener(_tabChanged);
+  }
 
-    final colorScheme = ColorScheme.of(context);
+  @override
+  void dispose() {
+    _tab.dispose();
+    super.dispose();
+  }
 
-    InlineSpan bold(String text) => TextSpan(
-      style: const TextStyle().merge(weightVariableTextStyle(context, wght: 700)),
-      text: text);
+  void _tabChanged() {
+    setState(() {
+      // The actual state lives in [_tab].
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    const pageBodies = [
+      (HomePageTab.inbox,          InboxPageBody()),
+      (HomePageTab.channels,       SubscriptionListPageBody()),
+      // TODO(#1094): Users
+      (HomePageTab.directMessages, RecentDmConversationsPageBody()),
+    ];
 
-    int? testStreamId;
-    if (store.connection.realmUrl.origin == 'https://chat.zulip.org') {
-      testStreamId = 7; // i.e. `#test here`; TODO cut this scaffolding hack
+    _NavigationBarButton button(HomePageTab tab, IconData icon) {
+      return _NavigationBarButton(icon: icon,
+        selected: _tab.value == tab,
+        onPressed: () {
+          _tab.value = tab;
+        });
     }
 
+    // TODO(a11y): add tooltips for these buttons
+    final navigationBarButtons = {
+      button(HomePageTab.inbox,          ZulipIcons.inbox),
+      button(HomePageTab.channels,       ZulipIcons.hash_italic),
+      // TODO(#1094): Users
+      button(HomePageTab.directMessages, ZulipIcons.user),
+      _NavigationBarButton(        icon: ZulipIcons.menu,
+        selected: false,
+        onPressed: () => _showMenu(context, tabNotifier: _tab)),
+    };
+
+    final designVariables = DesignVariables.of(context);
+    return Scaffold(
+      appBar: ZulipAppBar(titleSpacing: 16,
+        title: Text(_currentTabTitle)),
+      body: Stack(
+        children: [
+          for (final (tab, body) in pageBodies)
+            // TODO(#535): Decide if we find it helpful to use something like
+            //   [SemanticsProperties.namesRoute] to structure this UI better
+            //   for screen-reader software.
+            Offstage(offstage: tab != _tab.value, child: body),
+        ]),
+      bottomNavigationBar: DecoratedBox(
+        decoration: BoxDecoration(
+          border: Border(top: BorderSide(color: designVariables.borderBar, width: 0)),
+          color: designVariables.bgBotBar),
+        child: SafeArea(
+          // [Center] is necessary to limit the width of the row of icons,
+          // because it allows the child to be smaller than itself.
+          child: Center(
+            // This shrinks [Center] to be as tall as its child, instead of
+            // growing as tall as [Scaffold] allows, filling the entire screen.
+            heightFactor: 1,
+            child: ConstrainedBox(
+              // TODO(design): determine a suitable max width for bottom nav bar
+              constraints: const BoxConstraints(maxWidth: 600).tighten(height: 48),
```

Comment:
It's kept bugging me that this feels complicated, so I fiddled a bit to find a simpler version:
```suggestion
          child: SizedBox(height: 48,
            child: Center(
              child: ConstrainedBox(
                // TODO(design): determine a suitable max width for bottom nav bar
                constraints: const BoxConstraints(maxWidth: 600),
```

(The `CrossAxisAlignment.stretch` means the child will indeed fill those 48px of height, and therefore so will the Center widget)